templates/CMakeLists.j2

```diff
@@ -1,36 +1,29 @@
-cmake_minimum_required( VERSION 3.16 )
+cmake_minimum_required( VERSION 3.17 )
 project( {{ name }}
          LANGUAGES CXX )
 
+if( NOT WIN32 )
+    enable_language( C )
+endif()
+
 list( APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR} )
 
-SET(PYTHON_EXECUTABLE "" CACHE PATH "Pyhon executable")
-SET(PYTHON_LIBRARY "" CACHE PATH "Python library path - only relevant on windows")
-SET(PYTHON_SP_DIR "site-packages" CACHE PATH "Python site-packages directory")
+find_package (Python REQUIRED COMPONENTS Interpreter Development)
+message( STATUS "Python lib: ${Python_LIBRARIES}" )
 
-if(DEFINED ENV{CONDA_PREFIX})
-    if(WIN32)
-      set( PYTHON_EXECUTABLE "$ENV{CONDA_PREFIX}/python.exe" CACHE PATH "Pyhon executable" FORCE)
-      add_compile_definitions( _hypot=hypot _AMD64_ _WINDOWS)
-      execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "from sys import version_info as vi; print(f'{vi.major}{vi.minor}')"
-                      OUTPUT_VARIABLE PYTHON_VER )
-      string(STRIP ${PYTHON_VER} PYTHON_VER)
-      set( PYTHON_LIBRARY $ENV{CONDA_PREFIX}/libs/python${PYTHON_VER}.lib CACHE PATH "Python library path - only relevant on windows" FORCE)
-    else()
-      set( PYTHON_EXECUTABLE "$ENV{CONDA_PREFIX}/bin/python" CACHE PATH "Pyhon executble" FORCE)
-    endif()
-endif()
+SET(PYTHON_SP_DIR "site-packages" CACHE PATH "Python site-packages directory (for installing)")
 
 find_package( pybind11 REQUIRED )
-find_package( OpenCascade REQUIRED COMPONENTS OPENCASCADE )
-find_package( VTK 9 REQUIRED COMPONENTS WrappingPythonCore RenderingCore CommonDataModel CommonExecutionModel )
+find_package( VTK REQUIRED )
+find_package( OpenCASCADE REQUIRED )
 
 include_directories( ${PROJECT_SOURCE_DIR} )
 file( GLOB CPP_FILES ${PROJECT_SOURCE_DIR}/*.cpp )
 
 add_library( {{ name }} MODULE ${CPP_FILES} )
```

Comment:
Have you tried `pybind11_add_module` in place of this?

Reply:
Yes, I prefer to have explicit control on target creation.